packages/api/src/command/medical/patient/create-patient-settings.ts

```diff
@@ -0,0 +1,157 @@
+import { PatientSettings, PatientSettingsCreate } from "@metriport/core/domain/patient-settings";
+import { out } from "@metriport/core/util/log";
+import { uuidv7 } from "@metriport/core/util/uuid-v7";
+import { Transaction } from "sequelize";
+import { PatientModel } from "../../../models/medical/patient";
+import { PatientSettingsModel } from "../../../models/patient-settings";
+import { executeOnDBTx } from "../../../models/transaction-wrapper";
+import { getPatientIds, getPatientOrFail } from "./get-patient";
+
+/**
+ * Creates a new patient settings record
+ *
+ * @param patientId The patient ID
+ * @param cxId The customer ID
+ * @param adtSubscription Whether to enable or disable ADT subscription. Optional, defaults to false.
+ * @returns The created patient settings record
+ */
+export async function createPatientSettings({
+  patientId,
+  cxId,
+  adtSubscription,
+}: {
+  patientId: string;
+  cxId: string;
+  adtSubscription: boolean;
+}): Promise<PatientSettings> {
+  await getPatientOrFail({ cxId, id: patientId });
+
+  const patientSettingsCreate: PatientSettingsCreate = {
+    id: uuidv7(),
+    cxId,
+    patientId,
+    adtSubscription,
+  };
+
+  const newPatientSettings = await PatientSettingsModel.create(patientSettingsCreate);
+  return newPatientSettings;
+}
+
+/**
+ * Creates or updates patient settings for the given customer and patient IDs
+ *
+ * @param cxId The customer ID
+ * @param patientIds The patient IDs to update. Optional. If not provided, all patients for the CX will be updated.
+ * @param adtSubscription Whether to enable or disable ADT subscription. Optional, defaults to false.
+ * @returns The number of patients updated and the list of patients not found
+ */
+export async function createOrUpdatePatientSettings({
+  cxId,
+  patientIds = [],
+  adtSubscription = false,
+}: {
+  cxId: string;
+  patientIds?: string[];
+  adtSubscription?: boolean;
+}): Promise<{
+  patientsNotFound: string[];
+  patientsUpdated: number;
+}> {
+  const { log } = out(`createOrUpdatePatientSettings - cx ${cxId}`);
+
+  const idsToProcess = Array.isArray(patientIds) ? patientIds : [];
+  let finalIds = idsToProcess.length > 0 ? idsToProcess : await getPatientIds({ cxId });
```

Comment:
Why creating this as a variable? We can easily create a const here and a new array to get the resulting, validated patient IDs, so we can make the code more immutable (deterministic).